substrate/utils/wasm-builder/src/wasm_project.rs

```diff
@@ -697,69 +754,77 @@ fn build_project(
-			.mvp_features_only()
-			.debug_info(true)
-			.add_pass(wasm_opt::Pass::StripDwarf)
-			.run(&in_path, &wasm_compact_path)
-			.expect("Failed to compact generated WASM binary.");
-
-		let wasm_compact_compressed_path =
-			project.join(format!("{}.compact.compressed.wasm", out_name));
-		if compress_wasm(&wasm_compact_path, &wasm_compact_compressed_path) {
-			(Some(WasmBinary(wasm_compact_path)), Some(WasmBinary(wasm_compact_compressed_path)))
-		} else {
-			(Some(WasmBinary(wasm_compact_path)), None)
-		}
-	} else {
-		(None, None)
-	};
+	let wasm_compact_path = project.join(format!("{}.compact.wasm", out_name));
+	let start = std::time::Instant::now();
+	wasm_opt::OptimizationOptions::new_opt_level_0()
+		.mvp_features_only()
+		.debug_info(true)
+		.add_pass(wasm_opt::Pass::StripDwarf)
+		.run(&in_path, &wasm_compact_path)
+		.expect("Failed to compact generated WASM binary.");
+	println!(
+		"{} {}",
+		colorize_info_message("Compacted wasm in"),
+		colorize_info_message(format!("{:?}", start.elapsed()).as_str())
+	);
+	Some(WasmBinary(wasm_compact_path))
+}
+
+fn copy_bloaty_blob(project: &Path, inner_profile: &Profile, out_name: &str) -> WasmBinaryBloaty {
+	let in_path = project
+		.join("target/wasm32-unknown-unknown")
+		.join(inner_profile.directory())
+		.join(format!("{}.wasm", out_name));
 
 	let bloaty_path = project.join(format!("{}.wasm", out_name));
 	fs::copy(in_path, &bloaty_path).expect("Copying the bloaty file to the project dir.");
-
-	(wasm_compact_path, wasm_compact_compressed_path, WasmBinaryBloaty(bloaty_path))
+	WasmBinaryBloaty(bloaty_path)
 }
 
-fn compress_wasm(wasm_binary_path: &Path, compressed_binary_out_path: &Path) -> bool {
+fn try_compress_blob(compact_blob_path: &Path, out_name: &str) -> Option<WasmBinary> {
 	use sp_maybe_compressed_blob::CODE_BLOB_BOMB_LIMIT;
 
-	let data = fs::read(wasm_binary_path).expect("Failed to read WASM binary");
+	let project = compact_blob_path.parent().expect("blob path should have a parent directory");
+	let compact_compressed_blob_path =
+		project.join(format!("{}.compact.compressed.wasm", out_name));
+
+	let start = std::time::Instant::now();
+	let data = fs::read(compact_blob_path).expect("Failed to read WASM binary");
 	if let Some(compressed) = sp_maybe_compressed_blob::compress(&data, CODE_BLOB_BOMB_LIMIT) {
-		fs::write(compressed_binary_out_path, &compressed[..])
+		fs::write(&compact_compressed_blob_path, &compressed[..])
 			.expect("Failed to write WASM binary");
 
-		true
+		println!(
+			"{} {}",
+			colorize_info_message("Compressed blob in"),
+			colorize_info_message(format!("{:?}", start.elapsed()).as_str())
+		);
+		Some(WasmBinary(compact_compressed_blob_path))
 	} else {
 		build_helper::warning!(
-			"Writing uncompressed wasm. Exceeded maximum size {}",
+			"Writing uncompressed blob. Exceeded maximum size {}",
 			CODE_BLOB_BOMB_LIMIT,
 		);
-
-		false
+		println!("{}", colorize_info_message("Skipping blob compression"));
```

Comment:
Do we need this `println` here, considering just before it a warning is emitted?

Reply:
Good point. Will remove